python/main.py

```diff
@@ -86,11 +139,36 @@ async def get_image(image_name):
 
 @app.get("/items/{item_id}")
 def get_item(item_id: int):
-    with open(json_file, mode='r') as j:
-        items = json.load(j)
+    # connect to db
+    conn = sqlite3.connect(db)
+    # with open(json_file, mode='r') as j:
+    #     items = json.load(j)
 
     if item_id >= len(items['items']):
         raise HTTPException(status_code=404, detail="No item found with this id")
     
     else: 
+        # close db
+        conn.close()
         return items['items'][item_id - 1]
+
+
+@app.get("/search")
+def search_item(keyword: str):
+    logger.info(f"Search item name: {keyword}")  # keyword探す
+
+    # connect to db
+    conn = sqlite3.connect(db)
+    cur = conn.cursor()
+
+    # 検索
+    cur.execute('SELECT * FROM items WHERE name LIKE ?', (keyword,))
```

Comment:
Does this really performs the partial match, as suggested in the instruction?

> 指定したキーワードを含む商品一覧を返す

Reply:
I rewroted (keyword) to (f'%{keyword}%',).